src/mediapipe_internal/mediapipegraphexecutor.cpp

```diff
@@ -890,4 +890,131 @@ Status MediapipeGraphExecutor::infer(const KFSRequest* request, KFSResponse* res
+        }                                                                                 \
+    }
+
+template <typename T>
+class HolderWithRequestOwnership : public ::mediapipe::packet_internal::Holder<T> {
+    std::shared_ptr<const ::inference::ModelInferRequest> req;
+
+public:
+    explicit HolderWithRequestOwnership(const T* ptr, const std::shared_ptr<const ::inference::ModelInferRequest>& req) :
+        ::mediapipe::packet_internal::Holder<T>(ptr),
+        req(req) {}
+};
+
+// TODO: Add support for other types CVS-122328/CVS-122329
+Status MediapipeGraphExecutor::partialDeserialize(std::shared_ptr<const ::inference::ModelInferRequest> request, ::mediapipe::CalculatorGraph& graph) {
+    if (!request->id().empty()) {
+        auto requestTimestamp = stoi64(request->id());  // TODO: Decide if deserialize from id or int parameter
+        if (requestTimestamp.has_value()) {
+            currentStreamTimestamp = ::mediapipe::Timestamp(requestTimestamp.value());
+        } else {
+            return StatusCode::MEDIAPIPE_INVALID_TIMESTAMP;
+        }
+    }
+    for (const auto& input : request->inputs()) {
+        std::unique_ptr<ov::Tensor> tensor;
+        OVMS_RETURN_ON_FAIL(deserializeTensor(input.name(), *request, tensor), "ov::Tensor deserialization");
+        MP_RETURN_ON_FAIL(graph.AddPacketToInputStream(
+                              input.name(),
+                              ::mediapipe::packet_internal::Create(
+                                  new HolderWithRequestOwnership<ov::Tensor>(
+                                      tensor.release(),
+                                      request))
+                                  .At(currentStreamTimestamp)),
+            "adding packet to input stream", StatusCode::MEDIAPIPE_GRAPH_ADD_PACKET_INPUT_STREAM);
+    }
+    currentStreamTimestamp = currentStreamTimestamp.NextAllowedInStream();
+    return StatusCode::OK;
+}
+
+Status MediapipeGraphExecutor::inferStream(const ::inference::ModelInferRequest& firstRequest, ::grpc::ServerReaderWriterInterface<::inference::ModelStreamInferResponse, ::inference::ModelInferRequest>& stream) {
+    try {
+        // Init
+        ::mediapipe::CalculatorGraph graph;
+        MP_RETURN_ON_FAIL(graph.Initialize(this->config), "graph initialization", StatusCode::MEDIAPIPE_GRAPH_INITIALIZATION_ERROR);
+
+        // Installing observers
+        for (const auto& name : this->outputNames) {
+            MP_RETURN_ON_FAIL(graph.ObserveOutputStream(name, [&stream, &name](const ::mediapipe::Packet& packet) -> absl::Status {
+                try {
+                    ::inference::ModelStreamInferResponse resp;
+                    // TODO: Add support for other types CVS-122327/CVS-122329
+                    OVMS_RETURN_MP_ERROR_ON_FAIL(receiveAndSerializePacket<ov::Tensor>(packet, *resp.mutable_infer_response(), name), "ov::Tensor serialization");  // TODO: Missing test
```

Comment:
I was not able to mock behavior so that serialization fails in OVMS level:
- pushing wrong packet type from Process() method causes graph error, packet is not even delivered via callback

Pushing correct packet but not supported by serialization would be good test, but we do not have complete serialization yet https://github.com/openvinotoolkit/model_server/pull/2097